api/src/main/java/com/epam/pipeline/manager/cloud/gcp/GCPVMService.java

```diff
@@ -82,22 +87,40 @@ public Instance getRunningInstanceByRunId(final GCPRegion region, final String r
         }
     }
 
-    public Optional<InstanceTerminationState> getTerminationState(final GCPRegion region,
-                                                                  final String instanceId) {
+    public Optional<InstanceTerminationState> getTerminationState(final GCPRegion region, final String instanceId) {
+        Optional<InstanceTerminationState> result = Optional.empty();
         try {
-            final Instance instance = getInstanceById(region, instanceId);
-            if (instance != null && instance.getStatus().equals(GCPInstanceStatus.TERMINATED.name())) {
-                return Optional.of(
-                        InstanceTerminationState.builder()
-                                .instanceId(instanceId)
-                                .stateCode(GCPInstanceStatus.TERMINATED.name())
-                                .stateMessage(instance.getStatusMessage()).build()
-                );
+            final OperationList operationList = gcpClient.buildComputeClient(region)
+                    .zoneOperations()
+                    .list(region.getProject(), region.getRegionCode())
+                    .setFilter(String.format(COMPUTE_OPERATIONS_FILTER, instanceId)).execute();
+
+            if (!operationList.isEmpty()) {
```

Comment:
To reduce code indentation: `ListUtils.emptyIfNull(operationList.getItems()).fillter(...).findFirst(...).map(...)`

Reply:
Agree, looks much better :)